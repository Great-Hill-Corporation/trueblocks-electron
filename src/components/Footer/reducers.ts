import { Reducer } from 'redux';
import { Action, ActionCreator } from 'redux';

//-------------------------------------------------------------------
export const INCREMENT = 'foot/INCREMENT';
export const DECREMENT = 'foot/DECREMENT';
export const RESET = 'foot/RESET';

//-------------------------------------------------------------------
export interface IncrementAction extends Action {
  type: 'foot/INCREMENT';
}
export interface DecrementAction extends Action {
  type: 'foot/DECREMENT';
}
export interface ResetAction extends Action {
  type: 'foot/RESET';
}

//-------------------------------------------------------------------
export const increment: ActionCreator<IncrementAction> = () => ({
  type: INCREMENT
});
export const decrement: ActionCreator<DecrementAction> = () => ({
  type: DECREMENT
});
export const reset: ActionCreator<ResetAction> = () => ({
  type: RESET
});

//-------------------------------------------------------------------
export type FooterAction = IncrementAction | DecrementAction | ResetAction;

//-------------------------------------------------------------------
export interface FooterState {
  readonly footValue: number;
}
const defaultState: FooterState = {
  footValue: 10
};

//-------------------------------------------------------------------
export const footerReducer: Reducer<FooterState> = (state = defaultState, action: FooterAction) => {
  switch (action.type) {
    case INCREMENT:
      return {
        ...state,
        footValue: state.footValue + 1
      };
    case DECREMENT:
      return {
        ...state,
        footValue: state.footValue - 1
      };
    case RESET:
      return {
        ...state,
        footValue: defaultState.footValue
      };
    default:
      return state;
  }
};
