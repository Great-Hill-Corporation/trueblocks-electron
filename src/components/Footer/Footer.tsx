import * as React from 'react';
import { connect } from 'react-redux';
import { Dispatch, Action } from 'redux';
import { RootState } from '../../root-reducers';
import { FooterAction, decrement, increment, reset } from './reducers';

require('./styles.scss');

export interface FooterProps {
  footValue: number;
  incrementValue: () => Action;
  decrementValue: () => Action;
  resetValue: () => Action;
}

const FooterInner: React.FunctionComponent<FooterProps> = ({
  footValue,
  incrementValue,
  decrementValue,
  resetValue
}: FooterProps) => (
  <div className="primary">
    <p id="footer-value">Current footValue: {footValue}</p>
    <p>
      <button id="increment" onClick={incrementValue}>
        Increment
      </button>
      <button id="decrement" onClick={decrementValue}>
        Decrement
      </button>
      <button id="reset" onClick={resetValue}>
        Reset
      </button>
    </p>
  </div>
);

const mapStateToProps = (state: RootState) => ({
  footValue: state.footer.footValue
});

const mapDispatchToProps = (dispatch: Dispatch<FooterAction>) => ({
  incrementValue: () => dispatch(increment()),
  decrementValue: () => dispatch(decrement()),
  resetValue: () => dispatch(reset())
});

export const Footer = connect(mapStateToProps, mapDispatchToProps)(FooterInner);
