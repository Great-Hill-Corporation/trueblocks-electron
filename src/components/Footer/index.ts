import { Footer } from './Footer';
import { FooterAction, footerReducer } from './reducers';

//export type RootActions = FooterAction[keyof FooterAction];
export { Footer, FooterAction, footerReducer };
