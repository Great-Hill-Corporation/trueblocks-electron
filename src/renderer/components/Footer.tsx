import * as React from 'react';
import { connect } from 'react-redux';
import { Dispatch, Action } from 'redux';
import { RootState } from '../../root-reducers';
import { FooterAction, decrement, increment, reset } from '../actions/footerActions';

require('./Footer.scss');

export interface Props {
  value: number;
  incrementValue: () => Action;
  decrementValue: () => Action;
  resetValue: () => Action;
}

const Footer: React.FunctionComponent<Props> = ({ value, incrementValue, decrementValue, resetValue }: Props) => (
  <div className="footer">
    <p id="footer-value">Current value: {value}</p>
    <p>
      <button id="increment" onClick={incrementValue}>
        Increment
      </button>
      <button id="decrement" onClick={decrementValue}>
        Decrement
      </button>
      <button id="reset" onClick={resetValue}>
        Reset
      </button>
    </p>
  </div>
);

const mapStateToProps = (state: RootState) => ({
  value: state.footer.value
});

const mapDispatchToProps = (dispatch: Dispatch<FooterAction>) => ({
  incrementValue: () => dispatch(increment()),
  decrementValue: () => dispatch(decrement()),
  resetValue: () => dispatch(reset())
});

export default connect(mapStateToProps, mapDispatchToProps)(Footer);
