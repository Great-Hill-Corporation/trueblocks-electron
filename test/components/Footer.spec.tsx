import * as React from 'react';
import * as renderer from 'react-test-renderer';
import Footer from '../../src/renderer/components/Footer';

describe('Footer component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Footer value={1} incrementValue={jest.fn()} decrementValue={jest.fn()} resetValue={jest.fn()} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
