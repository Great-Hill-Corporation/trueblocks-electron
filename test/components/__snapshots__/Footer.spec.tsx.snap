// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component renders correctly 1`] = `
<div
  className="footer"
>
  <p>
    <img
      src="test-file-stub"
    />
  </p>
  <p
    id="footer-value"
  >
    Current value: 
    1
  </p>
  <p>
    <button
      id="increment"
      onClick={[MockFunction]}
    >
      Increment
    </button>
    <button
      id="decrement"
      onClick={[MockFunction]}
    >
      Decrement
    </button>
    <button
      id="reset"
      onClick={[MockFunction]}
    >
      Reset
    </button>
  </p>
</div>
`;
